src/numerics/trilinos_epetra_matrix.C

```diff
@@ -97,14 +97,14 @@ void EpetraMatrix<T>::update_sparsity_pattern (const SparsityPattern::Graph & sp
   if (m==0)
     return;
 
-  _graph = new Epetra_CrsGraph(Copy, *_map, &n_nz_tot[0]);
+  _graph = new Epetra_CrsGraph(Copy, *_map, n_nz_tot.data());
 
   // Tell the matrix about its structure.  Initialize it
   // to zero.
   for (numeric_index_type i=0; i<n_rows; i++)
     _graph->InsertGlobalIndices(_graph->GRID(i),
                                 cast_int<numeric_index_type>(sparsity_pattern[i].size()),
-                                const_cast<int *>((const int *)&sparsity_pattern[i][0]));
+                                const_cast<int *>(sparsity_pattern[i].data()));
```

Comment:
This broke compilation with Trilinos.  const_cast won't handle the unsigned->signed conversion.

Reply:
Oops, there was a hidden `reinterpret_cast` cast happening there, eh. I can fix that.